src/TLuaInterpreter.cpp

```diff
@@ -4195,45 +4195,44 @@ int TLuaInterpreter::setRoomIDbyHash(lua_State* L)
 {
     int id;
     if (!lua_isnumber(L, 1)) {
-        lua_pushstring(L, "setRoomIDbyHash: wrong argument type");
+        lua_pushfstring(L, "setRoomIDbyHash: bad argument #1 type (room id as number expected, got %s!)", luaL_typename(L, 1));
         lua_error(L);
         return 1;
     } else {
```

Comment:
We can also remove the else-after-return for this `if` as well - i.e. we probably want:
```
    if (!lua_isnumber(L, 1)) {
        lua_pushfstring(L, "setRoomIDbyHash: bad argument #1 type (room id as number expected, got %s!)", luaL_typename(L, 1));
        return lua_error(L);
    }
    int id = lua_tonumber(L, 1);
```

As it happens `lua_error(L);` does NOT return (it invokes the second part of the **[`setjmp`/`longjmp`](https://en.wikipedia.org/wiki/Setjmp.h)** C functions) and the `return` statement is ***never*** executed - some code analysers get confused by this and you will normally set that in *more up to date Mudlet code* we use the `return lua_error(L)` form which stops them getting confused...